versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5775 @@
+|`Map[String,Y]`|`Struct`|All `Struct` members must be coercible to `Y`|
+|`Object`|`Struct`||
+|`Struct`|`Object`|`Object` keys must match `Struct` member names, and `Object` values must be coercible to `Struct` member types|
+
+Note that primitive types can always be converted to `String` using [string interpolation](#expression-placeholders-and-string-interpolation).
+
+```wdl
+Int i = 5
+String istring = "~{i}"
+```
+
+See [Expression Placeholder Coercion](#expression-placeholder-coercion) for details.
+
+###### Coercion of Optional Types
+
+A non-optional type `T` can always be coerced to an optional type `T?`, but the reverse is not true - coercion from `T?` to `T` is not allowed because the latter cannot accept `None`.
+
+This constraint propagates into compound types. For example, an `Array[T?]` can contain both optional and non-optional elements. This facilitates the common idiom `select_first([expr, default])`, where `expr` is of type `T?` and `default` is of type `T`, for converting an optional type to a non-optional type. However, an `Array[T?]` could not be passed to the [`sep`](#-string-sepstring-arraystring) function, which requires an `Array[T]`.
+
+The two exceptions where coercion from `T?` to `T` are allowed are:
+* [String concatenation in expression placeholders](#concatenation-of-optional-values)
+* [Equality and inequality comparisons](#equality-and-inequality-comparison-of-optional-types)
+
+###### Struct/Object coercion from Map
+
+`Struct`s and `Object`s can be coerced from map literals, but beware the following behavioral difference:
+
+```wdl
+String a = "beware"
+String b = "key"
+String c = "lookup"
+
+# What are the keys to this object?
+Object object_syntax = object {
+  a: 10,
+  b: 11,
+  c: 12
+}
+
+# What are the keys to this object?
+Object map_coercion = {
+  a: 10,
+  b: 11,
+  c: 12
+}
+```
+
+- If an `Object` is specified using the object-style `Object map_syntax = object { a: ...` syntax then the keys will be `"a"`, `"b"` and `"c"`.
+- If an `Object` is specified using the map-style `Object map_coercion = { a: ...` then the keys are expressions, and thus `a` will be a variable reference to the previously defined `String a = "beware"`.
+
+### Fully Qualified Names & Namespaced Identifiers
+
+```txt
+$fully_qualified_name = $identifier ('.' $identifier)*
+$namespaced_identifier = $identifier ('.' $identifier)*
+```
+
+A fully qualified name is the unique identifier of any particular call, input, or output, and has the following structure:
+
+* For calls: `<parent namespace>.<call alias>`
+* For inputs and outputs: `<parent namespace>.<input or output name>`
+
+The `parent namespace` here will equal the fully qualified name of the workflow containing the call, or the workflow or task containing the input or output. For the top-level workflow this is equal to the workflow name.
+
+For example:
```

Comment:
Preface this with a link to the text about workflow imports first.

Reply:
I think this section is out of place. I'll move it to after discussing imports.